Modules/CTP/src/RawDataReaderCheck.cxx

```diff
@@ -37,7 +38,57 @@ using namespace o2::ctp;
 namespace o2::quality_control_modules::ctp
 {
 
-void RawDataReaderCheck::configure() {}
+void RawDataReaderCheck::configure()
+{
+  float casted;
+  std::string param = mCustomParameters["thresholdRateBad"];
+  if (isdigit(param[0])) {
```

Comment:
this can work only by some luck. you are checking if the first character in the string is a digit. Thus it will fail if it is e.g. "-1.0" or ".3" which are valid floating point numbers. Needless to say, it will also collapse for cases like "1xyz".

I think you will have to rely on handling exceptions thrown by `std::stof`, i don't know a more elegant way in C++ :/
I would propose to try this:
```
try {
  mThresholdRateBad = std::stof(mCustomParameters["thresholdRateBad"]);
} catch (const std::exception& e) {
  mThresholdRateBad = 0.15;
}
```

Reply:
The case like  "1xyz", I tried and it does not fail. The casted value is then 1. But you are right, that I didn't think about cases like "-1.0" or ".3". In this case a negative value doesn't make sense, so my check would be ok. So I will do the exception check + logic check that the threshold value should be 0 < thr < 1.